javascript/javascript3/week2/currencyCalculator.js

```diff
@@ -0,0 +1,151 @@
+if (inputSaved) {
+    currencyFromInput.value = inputSaved["country-from"];
+    currencyToInput.value = inputSaved["country-to"];    
+    displayCurrencyFrom.innerHTML = `1 ${currencyFromInput.value} equals`;
+}
+//-------------------------------------------------//
+// STEP 1: FETCH DATA FROM API
+//Fetch Data from API
+const exchangeRatesData = async () => {
+
+    // When the convert button is clicked, calculate currency exchange.
+    let baseCurrency = currencyFromInput.value;
+   
+    // If location is blank, then default currency = "EUR"
+    if (baseCurrency == "") {
+        console.log("True");
+        baseCurrency = "EUR"
+    } else {
+        baseCurrency = currencyFromInput.value;        
+    }
+    
+    try {
+        const response = await fetch(`${BASE_URL}/v6/${API_KEY}/latest/${baseCurrency}`)
+        console.log(response.status);  //200 (Should be between 200 and 299)
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.log(error)
+    }
+
+};
+
+// STEP 2: SAVE DATA TO LOCAL STORAGE
+const saveToLocalStorage = async () => {
+    const data = await exchangeRatesData();
+
+    const exchangeData = await (data["conversion_rates"])
+    let countryCode = Object.keys(data["conversion_rates"])
+    
+    let rate = Object.values(data["conversion_rates"])
+   
+    const exchangeInfo = {
+        "countryCode": countryCode,
+        "rate": rate
+    }
+    
+    const input = {
+        "country-from": "EUR",
+        "country-to": "DKK"
+    }
+
+    // Save to Local Storage
+    localStorage.setItem("exchangeInfo", JSON.stringify(exchangeInfo))
+    localStorage.setItem("input", JSON.stringify(input))
+
+} // End of function saveToLocalStorage
+
+saveToLocalStorage()
+
+//------------------------------------------------//
+
+// Extract Data - To get exchange rates
+const display = async (event) => {
+    const data = await exchangeRatesData().then((data) => {
+        
+        // To display currency data at top of page
+        displayCurrencyFrom.innerHTML = `1 ${currencyFromInput.value} equals`;        
+        displayCurrencyTo.innerHTML = (selectCountryTo.value);
+
+        // To get the country code of the country to
+        const countryToCode = selectCountryTo.value
+
+        // To get and display conversion rate
+        const conversionRate = ((data["conversion_rates"][countryToCode]).toFixed(2))
+        displayConversionRate.innerHTML = conversionRate;
+
+        // To get the amount value entered in country from
+        // To add amount to convert -> 1 or 10 etc
+        const amountFrom = document.querySelector("#amount-from").value;
+        console.log (amountFrom);
+
+        // To display calculated amount when the number in base currency is changed
+        displayConvertedAmount.value = amountFrom * conversionRate;
+        
+        // To display Date and Time:
+        displayDateAndTime.innerHTML = new Date();
```

Comment:
Nicely done. Try exploring the [date methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) to convert it into the string you want.

Reply:
Thank you